src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -122,6 +125,17 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 								  "foreign tables.")));
 	}
 
+	/* check that the relation does not contain any rows */
+	if (!LocalTableEmpty(distributedRelationId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_TABLE_DEFINITION),
+						errmsg("cannot distribute relation \"%s\"",
+							   distributedRelationName),
+						errdetail("Relation \"%s\" contains data.",
+								  distributedRelationName),
+						errhint("Please empty your table before distributing it")));
```

Comment:
I think there is no need to say `Please`, I don't think it is common to have it in the error messages.
And, the sentence should end with a period.